sphinx/util/inspect.py

```diff
@@ -23,8 +25,25 @@
 memory_address_re = re.compile(r' at 0x[0-9a-f]{8,16}(?=>$)')
 
 
+def isgeneratorfunction(func):
+    """Like inspect.isgeneratorfunction but supports functools.partial."""
+    try:
+        if inspect.ismethod(func):
+            func = func.__func__
+        if type(func) is partial:
+            func = func.func
+        return inspect.isgeneratorfunction(func)
+    except AttributeError:
+        logging.warn('This fail is weird, is %r a generator?', func)
```

Comment:
We don't use `logging` anywhere else.  Could this just bubble up and then use the ubiquitous `warnfunc`, or `warnings.warn`?